include/emitc/emitc_tosa.h

```diff
@@ -120,9 +120,6 @@ Dest conv2d(Src input, Weights weights, Tensor1D<int64_t, 4> padding,
   static_assert(is_tensor_of_dim<4, Weights>::value,
                 "Expected 4 dimensional weights");
 
-  assert(stride[0] == 1);
-  assert(stride[1] == 1);
```

Comment:
Maybe
```c
assert(stride[0] > 0);
assert(stride[1] > 0);
```

to be on the safe side?

Reply:
Side note, not relevant to this PR: Should the base type of `stride` be an unsigned int?

Reply:
These are signless integers on the mlir side, which we emit as signed integers in C++. I'm not sure if we want to have implicit conversions going on there.

Reply:
@simon-camp I dont think this check is needed, as this only would occur with invalid IR?
@marbre Speaking about tosa.conv2d: TOSA MLIR Dialect specifies strides as 64 bit integer array. Logically you're right.

Reply:
> These are signless integers on the mlir side, which we emit as signed integers in C++. I'm not sure if we want to have implicit conversions going on there.

Yeah, that was the question. I am not sure either, that's why I asked.

Reply:
I added assert statements to check that the strides are positive.